frontend/src/container/FormAlertRules/BasicInfo.tsx

```diff
@@ -137,32 +159,64 @@ function BasicInfo({
 					name="alert_all_configured_channels"
 					label="Alert all the configured channels"
 				>
-					<Switch
-						checked={shouldBroadCastToAllChannels}
-						onChange={handleBroadcastToAllChannels}
-					/>
+					<Tooltip
+						title={
+							noChannels
+								? 'Create a notification channel to configure alert'
+								: undefined
+						}
+						placement="right"
+					>
+						<Switch
+							checked={shouldBroadCastToAllChannels}
+							onChange={handleBroadcastToAllChannels}
+							disabled={noChannels || !!channels.loading}
+						/>
+					</Tooltip>
 				</FormItemMedium>
 
 				{!shouldBroadCastToAllChannels && (
-					<FormItemMedium
-						label="Notification Channels"
-						name="notification_channels"
-						required
-						rules={[
-							{ required: true, message: requireErrorMessage(t('field_alert_name')) },
-						]}
+					<Tooltip
+						title={
+							noChannels
+								? 'Create a notification channel to configure alert'
+								: undefined
+						}
+						placement="right"
 					>
-						<ChannelSelect
-							disabled={shouldBroadCastToAllChannels}
-							currentValue={alertDef.preferredChannels}
-							onSelectChannels={(preferredChannels): void => {
-								setAlertDef({
-									...alertDef,
-									preferredChannels,
-								});
-							}}
-						/>
-					</FormItemMedium>
+						<FormItemMedium
+							label="Notification Channels"
+							name="notification_channels"
+							required
+							rules={[
+								{ required: true, message: requireErrorMessage(t('field_alert_name')) },
+							]}
+						>
+							<ChannelSelect
+								disabled={
+									shouldBroadCastToAllChannels || noChannels || !!channels.loading
+								}
+								currentValue={alertDef.preferredChannels}
+								channels={channels}
+								onSelectChannels={(preferredChannels): void => {
+									setAlertDef({
+										...alertDef,
+										preferredChannels,
+									});
+								}}
+							/>
+						</FormItemMedium>
+					</Tooltip>
+				)}
+
+				{noChannels && addNewChannelPermission && (
+					<Button
+						onClick={handleCreateNewChannels}
+						icon={<PlusOutlined />}
+						className="create-notification-btn"
+					>
+						Create a notification channel
+					</Button>
```

Comment:
When there are no channels and the user is an editor, I think it makes sense to add a note that only the admin can create a new channel and nudge them to get the channel created. See this conversation https://app.intercom.com/a/inbox/wmkurvb7/inbox/conversation/3561

Reply:
Yeah, rather than hiding we can do some improvements here to better communicate with users to let them know what is happening here.

the current implementation is what we had on the alert channels page, I think we should improve it there also.

Reply:
I highly suggest we do it because we get support questions asking the same. I will leave it to you whether to address it in this PR or a separate PR.

Reply:
This looks good, made the change to both this page and alert channels page

<img width="1263" alt="image" src="https://github.com/SigNoz/signoz/assets/162284829/2f90aa54-814a-4786-8b4a-2929f1e05af7">

<img width="1268" alt="image" src="https://github.com/SigNoz/signoz/assets/162284829/d04a7ec4-04d9-4a53-90c9-f63d970e1c07">

<img width="1659" alt="image" src="https://github.com/SigNoz/signoz/assets/162284829/214435bc-318e-4ecb-922a-095783aa2e9c">